src/main/java/com/artipie/rpm/hm/IsXmlEqual.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.hm;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.cactoos.scalar.Unchecked;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.xmlunit.matchers.CompareMatcher;
+
+/**
+ * Matcher for xmls.
+ * @since 0.10
+ */
+public final class IsXmlEqual extends TypeSafeMatcher<Path> {
+
+    /**
+     * Body matcher.
+     */
+    private final byte[] xml;
+
+    /**
+     * Ctor.
+     * @param xml Xml bytes
+     */
+    @SuppressWarnings("PMD.ArrayIsStoredDirectly")
+    public IsXmlEqual(final byte[] xml) {
+        this.xml = xml;
+    }
+
+    /**
+     * Ctor.
+     * @param xml Xml string
+     */
+    public IsXmlEqual(final String xml) {
+        this(xml.getBytes());
+    }
+
+    /**
+     * Ctor.
+     * @param xml Xml file
+     */
+    public IsXmlEqual(final Path xml) {
+        this(new Unchecked<>(() -> Files.readAllBytes(xml)).value());
+    }
+
+    @Override
+    public boolean matchesSafely(final Path item) {
+        return CompareMatcher
+            .isIdenticalTo(item.toFile())
+            .ignoreWhitespace().normalizeWhitespace().matches(this.xml);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.StringInstantiation")
+    public void describeTo(final Description description) {
+        description.appendText(new String(this.xml));
```

Comment:
@olenagerasimova conversion of bytes into `String` might fail, as generally not every byte array can be converted to `String`. Though this scenario is unlikely, so I am not sure this really need fixing. Maybe if we specify `US_ASCII` encoding here we avoid exceptions in any case